uber/site_sections/registration.py

```diff
@@ -112,32 +112,33 @@ def index(self, session, message='', page='0', search_text='', uploaded_id='', o
         }  # noqa: E711
 
     @log_pageview
-    def form(self, session, message='', return_to='', check_in='', **params):
+    def form(self, session, message='', return_to='', **params):
         attendee = session.attendee(
             params, checkgroups=Attendee.all_checkgroups, bools=Attendee.all_bools, allow_invalid=True)
 
         if 'first_name' in params:
+            message = ''
+            
             attendee.group_id = params['group_opt'] or None
             if params.get('no_badge_num') or not attendee.badge_num:
-                attendee.badge_num = None
+                if params.get('save') == 'save_check_in' and attendee.badge_type not in c.PREASSIGNED_BADGE_TYPES:
+                    message = "Please enter a badge number to check this attendee in"
+                else:
+                    attendee.badge_num = None
 
             if 'no_override' in params:
                 attendee.overridden_price = None
 
-            message = ''
             if c.BADGE_PROMO_CODES_ENABLED and 'promo_code' in params:
                 message = session.add_promo_code_to_attendee(attendee, params.get('promo_code'))
 
             if not message:
-                
                 message = check(attendee)
 
             if not message:
                 # Free group badges are only considered 'registered' when they are actually claimed.
                 if attendee.paid == c.PAID_BY_GROUP and attendee.group_id and attendee.group.cost == 0:
                     attendee.registered = localized_now()
-                if check_in:
-                    attendee.checked_in = localized_now()
```

Comment:
I see you're deleting this, but don't we instead want to change this if-check to ``if params.get('save_check_in'):`` ?

As we have it now, I don't see where we'd be actually doing the check-in in this function.

Reply:
It's on line 162/163 -- there's a whole loop now that makes sure the changes that you make don't make an attendee unable to check in. The `check_in` parameter was no longer being passed to this page handler from the form, so we weren't hitting this line at all.

Reply:
Ahh, gotcha, I did a Ctrl+f to see if I'd missed it but I forgot to expand the code first.  In that case this looks good to go!